DirectProgramming/DPC++FPGA/Tutorials/Features/loop_fusion/src/loop_fusion.cpp

```diff
@@ -0,0 +1,280 @@
+      accessor accessor_array_1(buffer_array_1, h, write_only, no_init);
+      accessor accessor_array_2(buffer_array_2, h, write_only, no_init);
+
+      h.single_task<DefaultNoFusionKernel>([=
+      ]() [[intel::kernel_args_restrict]] {  // NO-FORMAT: Attribute
+        for (size_t j = 0; j < kN; j++) {
+          // Different tripcounts, does not fuse by default
+          for (size_t i = 0; i < kM + 1; i++) {
+            accessor_array_1[i] = i;
+          }
+          for (size_t i = 0; i < kM; i++) {
+            accessor_array_2[i] = accessor_array_1[i];
+          }
+        }
+
+      });
+    });
+
+    auto kernel_time = KernelRuntime(e);
+
+    // kernel consists of two loops with one array assignment in each.
+    int num_ops_per_kernel = kN * (2 * kM + 1);
+    std::cout << "Throughput for kernel without fusion by default and with "
+                 "arrays of sizes "
+              << kM + 1 << " and " << kM << ": "
+              << ((double)num_ops_per_kernel / kernel_time) << " Ops/ns\n";
+
+  } catch (sycl::exception const &e) {
+    ErrorReport(e);
+  }
+}
+
+// The compiler is explicitly told to fuse the inner loops using the
+// fpga_loop_fuse<>() function. Were this function not used, the loops would not
+// fuse by default, since the trip counts of the loops are different.
+void FusionFunction(std::array<int, kM + 1> &m_array_1,
+                    std::array<int, kM> &m_array_2) {
+  try {
+    queue q(selector, dpc_common::exception_handler,
+            property::queue::enable_profiling{});
+
+    buffer buffer_array_1(m_array_1);
+    buffer buffer_array_2(m_array_2);
+
+    event e = q.submit([&](handler &h) {
+      accessor accessor_array_1(buffer_array_1, h, write_only, no_init);
+      accessor accessor_array_2(buffer_array_2, h, write_only, no_init);
+
+      h.single_task<FusionFunctionKernel>([=
+      ]() [[intel::kernel_args_restrict]] {  // NO-FORMAT: Attribute
+        sycl::ext::intel::fpga_loop_fuse<2>([&] {
+          for (size_t j = 0; j < kN; j++) {
+            // Different tripcounts, does not fuse by default
+            for (size_t i = 0; i < kM + 1; i++) {
+              accessor_array_1[i] = i;
+            }
+            for (size_t i = 0; i < kM; i++) {
+              accessor_array_2[i] = accessor_array_1[i];
+            }
+          }
+        });
+      });
+    });
+
+    auto kernel_time = KernelRuntime(e);
+    // kernel consists of two loops with one array assignment in each.
+    int num_ops_per_kernel = kN * (2 * kM + 1);
+    std::cout << "Throughput for kernel with the loop fusion function wrapper "
+                 "and with arrays of sizes "
+              << kM + 1 << " and " << kM << ": "
+              << ((double)num_ops_per_kernel / kernel_time) << " Ops/ns\n";
+
+  } catch (sycl::exception const &e) {
+    ErrorReport(e);
+  }
+}
+
+int main() {
+  // Arrays will be populated in kernel loops
+  std::array<int, kM> default_fusion_1, default_fusion_2, no_fusion_1,
```

Comment:
Might be clearer to have different declaration lines for fusion, no_fusion, ...     Make the code so the output fo clang-format is the most readable.